### ISSUE TITLE
Population sum total differences exist between original codebase and foundation

### ISSUE THREAD

**Issue by bryce-carson** [KrishnamurthyLab/spatialEpisim.foundation#22]
### Original code algorithm
The algorithm the original code follows, as I recall it from our Tuesday Sept. 3rd meeting:
1. a spatial raster class is instantiated from a TIFF file
2. it is aggregated, if desired
3. it is cropped, if desired
4. it is resampled

### Foundation algorithm
The new code-base follows this algorithm, as I recall (being its author):
1. a spatial raster class is instantiated from a TIFF file
2. it is cropped, if desired
3. it is aggregated, if desired

# TODO
I need to determine what the largest cause of the population sum total difference is and rectify that. The new and the old code-base shouldn't differ on such a detail.

**Comment by bryce-carson** [comment]
This test validates that aggregating and cropping does not impact the population of the area(s) of interest.

- [ ] Remove or use the `layers` object in test of aggregation and cropping of population raster

https://github.com/KrishnamurthyLab/spatialEpisim.foundation/blob/6dd5f47b98eae4579f9c9a5798419d2aff7d80c2/tests/testthat/test-SVEIRD.BayesianDataAssimilation.R#L33-L42

**Comment by bryce-carson** [comment]
#27 is a duplicate of item 4 in this issue.

**Comment by bryce-carson** [comment]
I think I'll resolve this issue by adding the ability to toggle between resampling or not.

**Comment by bryce-carson** [comment]
Various resampling methods were tried in the original codebase, so having a drop-down to select the resampling method when resampling is enabled would be useful.

https://github.com/ashokkrish/spatialEpisim/blob/1b64d4538b5d0e4db14645859a3a600a13eed9bc/R/rasterStack.R#L69

**Comment by bryce-carson** [comment]
I have done some review of the old code and it appears that `terra::resample` was only used as part of an effort to crop the raster of the population to the silhouette of the nation under study. It transfers values from a source to a sink spatRaster, and the sink is the population spatRaster with the source being a binary rasterization (the values are either zero or one) of the provincial and national borders of the nation.

In conclusion, I don't need to worry about the fact that resample was used. It was used improperly, more or less, in the original code.
